cookbooks/ey-backup/recipes/mysql.rb

```diff
@@ -41,7 +41,13 @@
 	end
 end
 
-if node.dna['backup_window'] != 0 && ['db_master','solo'].include?(node.dna['instance_role'])
+is_backup_instance = if node['db_backup']['mysql']['backup_slave']
+  ['db_slave'].include?(node.dna['instance_role']) && node['db_backup']['mysql']['db_slave_name'] == node['dna']['name']
```

Comment:
I think this change means that the customer *must* specify the name of the replica which should run the backup.  What happens if they don't? What happens if the name they specify doesn't exist? Is it possible to gracefully fall back to the same behaviour as PostgreSQL in either of those cases, and just choose the first replica?

(If so then it would be good to have this behaviour for both, and simply set `default['db_backup']['mysql']['backup_slave']` true for PostgreSQL while leaving `default['db_backup']['mysql']['db_slave_name']` empty by default.)

Reply:
Obviously we would rename the variables if we were making this more general.
Perhaps:
`default['db_backup']['run_backup_on_replica']`
`default['db_backup']['backup_replica_name']`